src/platform-aws.c

```diff
@@ -317,6 +317,32 @@ int configure_nvls_option(void)
 	return 0;
 }
 
+int configure_net_flush(void)
+{
+	/* Hopper GPUs do not require a network flush, but NCCL versions <2.19.1
+	 * still enable flush by default on any GPU type.
+	 * For GPU generations earlier than Hopper, NCCL always enables flush, while
+	 * for Hopper GPUs flush is enabled or disabled depending on the value of
+	 * the NCCL_NET_FORCE_FLUSH environment variable. The default value for this
+	 * variable is 1 for NCCL versions <2.19.1, which forces flush when it is not
+	 * needed, so it is safe to set it to 0 if it is not explicitly set.
+	 */
+
+	int ret;
+
+	if ((NULL == getenv("NCCL_NET_FORCE_FLUSH")) &&
+		(0 == strcmp(get_platform_type(), "p5.48xlarge"))) {
```

Comment:
Follow the pattern in the rest of this code and make it a property in the instance type data structure.

Reply:
this instance type check is redundant as NCCL ignores the env variable on non-Hopper platforms anyway.
See [here](https://github.com/NVIDIA/nccl/blob/b6d7438d3145a619f924dbbca6c96db21fab716e/src/graph/paths.cc#L413C21-L413C33):

```
  // Flush is required on Ampere and earlier
  *flush = gpu->gpu.cudaCompCap < 90 ? 1 : ncclParamNetForceFlush();
```

But I would keep it to be safe, in case NCCL in the future changes this behavior. I agree, I will make it a property of the instance type struct.